Teleportation locomotion example

jmiskovic opened the pull request:
Teleporting locomotion combined with snap turning. 

Restricted to ground plane to keep it simple, some next example should have physics and geometry raycasting.

bjornbytes commented:
Thoughts on adding a fade out/in?  `motion` would need to manage some delay value which may complicate things too much.

jmiskovic commented:
Maybe there's a better way to do fading? I used the fill() and that depends on order of drawing.

jmiskovic commented:
Changes:
- There was a bug in teleport translation in combination with snap turning, fixed now
- Renamed example to "Teleportation_Flat"
- Changed disc accent color

I think this example is ready for merge.
